### ISSUE TITLE
Adding official py3.7 support

### ISSUE THREAD

**Issue by m1keil** [cloudreach/sceptre#622]
3.7 was released more than 6 months ago. 3.8 is scheduled to be released in October. Personally, I've been using Sceptre v2 with 3.7 for few months now without any issues. Is it time to add py37 to tox?

**Comment by ngfgrant** [comment]
Hey @m1keil if you could try add 3.7 to tox and run tests, if everything passes then I am happy to add official support for 3.7. If there are things that are breaking then for v2 we will support >=2.7-3.6 for now. Plan is to drop Py2 for v3.

**Comment by devenney** [comment]
I tinkered with this a bit when I was getting set up. For what it's worth, the only issues I experienced were with the test dependencies:

* `freezegun` has to be upgraded to 0.3.11.
* `moto` has to be upgraded to 1.3.7.

There is an outstanding deprecation warning coming from `PyYAML` which is covered by [PyYAML #202](https://github.com/yaml/pyyaml/issues/202). This either pollutes Sceptre output or requires silencing the deprecation warning in `sceptre/__init__.py` like so:

```
import warnings

with warnings.catch_warnings():
    warnings.filterwarnings("ignore", category=DeprecationWarning)
```

Suppose it comes down to whether you want to silence deprecation warnings in the project @ngfgrant, but Python 3.7 support seems functional. I've pushed the code I have (see above) for example.
